helix-core/src/test/java/org/apache/helix/integration/spectator/TestRoutingTableProviderPeriodicRefresh.java

```diff
@@ -203,34 +208,32 @@ synchronized int getRefreshCount() {
 
   @Test
   public void testPeriodicRefresh() throws InterruptedException {
-    // Wait so that initial refreshes finish (not triggered by periodic refresh timer)
-    Thread.sleep(1000L);
+    // Wait to ensure that the initial refreshes finish (not triggered by periodic refresh timer)
+    Thread.sleep(REFRESH_PERIOD_MS * 2);
 
     // Test short refresh
     int prevRefreshCount = _routingTableProvider.getRefreshCount();
-    // Wait for one timer duration
-    Thread.sleep(1000L);
-    // The timer should have gone off, incrementing the refresh count
-    Assert.assertEquals(_routingTableProvider.getRefreshCount(), prevRefreshCount + 1);
+    // Wait for more than one timer duration
+    Thread.sleep((long) (REFRESH_PERIOD_MS * 1.5));
+    // The timer should have gone off, incrementing the refresh count by one or two depends on the
+    // timing.
+    int newRefreshCount = _routingTableProvider.getRefreshCount();
+    Assert.assertTrue(
+        newRefreshCount == prevRefreshCount + 1 || newRefreshCount == prevRefreshCount + 2);
```

Comment:
While this eliminates flakiness, would it be too loose to ensure correctness?

Reply:
The java schedule task method that we are using is not accurate anyway. Moreover, the main test purpose here is to ensure the refresh happens. If it is faster a little bit, I guess we are still good.
On the other hand, if the refresh does respect the interval, so during 1.5 seconds, it happens more than 2 times, then the test will still fail.

Reply:
This comment is tied with my other comment. If we already know it's flaky between 1 or 2 refreshes, why don't we just `assert(+1 || +2)`, while keeping `sleep(1000)`? 

If we change it to `sleep(1500)`, we are opening doors to "slow refresh" cases. If a refresh doesn't respect the refresh period, and instead took around 1500, the previous version of the test will catch it, but now it won't. Like you said, maybe even 1500 is fine since there is no hard requirement, we are just ensuring the refresh happens, but it feels to me that correctness check is relaxed.

Reply:
As I replied to Lei, this PR is not to improve the test. This is mainly for unblocking the other devs since the test cannot pass. Alternatively, we can disable it. But I prefer the current version. At least it checks some of the logic.

We will need to address this by re-writing the test logic.

Reply:
Talked offline: since periodic refresh can happen slightly less, equally, or slightly more frequent than the configured wait period, I agree with @jiajunwang 's change to use both `sleep(1500)` and `assert(+1 || +2)`.